### PR TITLE
Add support for delete planning with limits in presence of foreign keys

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR makes the changes to support delete queries with limit along with foreign keys. After the changes in this PR, queries like the following now work - 
```SQL
delete from u_tbl10 order by id limit 1
```

Fuzzer tests for the same have been added and verified everything is working as intended.

## Related Issue(s)

- Part of https://github.com/vitessio/vitess/issues/12967

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by GuptaManan100** [comment]
The changes in this PR are based on top of https://github.com/vitessio/vitess/pull/15081 and should be merged after it.

File: go/test/endtoend/vtgate/foreignkey/fk_fuzz_test.go [hunk 1]
```diff
@@ -604,61 +608,65 @@ func TestFkFuzzTest(t *testing.T) {
...
+		//}, {
+		//	name:           "Single Thread - Balanced Inserts and Updates",
+		//	concurrency:    1,
+		//	timeForTesting: 5 * time.Second,
+		//	maxValForCol:   5,
+		//	maxValForId:    10,
+		//	insertShare:    50,
+		//	deleteShare:    0,
+		//	updateShare:    50,
+		//},
```
Comment (frouioui) [inline go/test/endtoend/vtgate/foreignkey/fk_fuzz_test.go:639]: We can probably remove this
```suggestion
```

File: go/test/endtoend/vtgate/foreignkey/fk_fuzz_test.go [hunk 2]
```diff
@@ -604,61 +608,65 @@ func TestFkFuzzTest(t *testing.T) {
...
+		//}, {
+		//	name:           "Multi Thread - Balanced Inserts, Updates and Deletes",
+		//	concurrency:    30,
+		//	timeForTesting: 5 * time.Second,
+		//	maxValForCol:   5,
+		//	maxValForId:    30,
+		//	insertShare:    50,
+		//	deleteShare:    50,
+		//	updateShare:    50,
+		//},
```
Comment (frouioui) [inline go/test/endtoend/vtgate/foreignkey/fk_fuzz_test.go:668]: same here
```suggestion
```

File: go/test/endtoend/vtgate/foreignkey/fk_fuzz_test.go (hunk 1 above)
Comment (frouioui) [reply go/test/endtoend/vtgate/foreignkey/fk_fuzz_test.go:639]: or add it back, i am unsure why it was commented in the first place

File: go/test/endtoend/vtgate/foreignkey/fk_fuzz_test.go (hunk 2 above)
Comment (harshit-gangal) [reply go/test/endtoend/vtgate/foreignkey/fk_fuzz_test.go:668]: oversight after testing. will put it back

File: go/vt/vtgate/planbuilder/operators/delete.go [hunk 3]
```diff
@@ -90,19 +98,58 @@ func createOperatorFromDelete(ctx *plancontext.PlanningContext, deleteStmt *sqlp
...
-		panic(vterrors.VT12001("foreign keys management at vitess with limit"))
-	}
 
 	return createFkCascadeOpForDelete(ctx, op, delClone, childFks, delOp.Target.VTable)
 }
 
+func deletePlanningForLimitFk(ctx *plancontext.PlanningContext, del *sqlparser.Delete) Operator {
+	delClone := ctx.SemTable.Clone(del).(*sqlparser.Delete)
+	del.Limit = nil
+	del.OrderBy = nil
```
Comment (frouioui) [inline go/vt/vtgate/planbuilder/operators/delete.go:112]: is it expected to modify the original deletect by setting `Limite` and `OrderBy` to `nil`?

Comment (harshit-gangal) [reply go/vt/vtgate/planbuilder/operators/delete.go:112]: select has the limit and order by. original query is removed and a where condition is added to provide the values for it from the select

**Review by harshit-gangal — APPROVED** [review]

**Pull request merged**